File: app/src/main/java/com/smartsoft/pereverzev/dagger2oldtutorial/screens/HomeActivityModule.java
package com.smartsoft.pereverzev.dagger2oldtutorial.screens;

import dagger.Module;
import dagger.Provides;

@Module
public class HomeActivityModule {

    private final HomeActivity homeActivity;

    public HomeActivityModule(HomeActivity homeActivity) {
        this.homeActivity = homeActivity;
    }

    @Provides
    @HomeActivityScope
    public HomeActivity homeActivity() {
        return homeActivity;
    }
}

File: app/src/main/kotlin/com/smartsoft/pereverzev/dagger2newtutorial/screens/HomeActivityModule.kt
package com.smartsoft.pereverzev.dagger2newtutorial.screens

import dagger.Module
import dagger.Provides

/**
 * Created by alexander on 03.01.2018.
 */
@Module
class HomeActivityModule(private val homeActivity: HomeActivity) {

    @Provides
    @HomeActivityScope
    fun homeActivity(): HomeActivity = homeActivity
}